### ISSUE TITLE
Control CPU affinity of Flutter threads, enforcing running on big cores (willing to contribute)

### ISSUE THREAD

**Issue by fzyzcjy** [flutter/flutter#105842]
## Use case

Hi thanks for the wonderful library! We know modern mobile phones have big and little cores. If a thread is run in little cores, it is much slower (though less power-hungry). We also know cpu-intensive libraries, such as some mobile neural network inference libs, all provide features to set CPU affinity to have better and more stable performance.

Therefore, I would appreciate it if the CPU affinity of flutter threads could be controlled. Maybe including the CPU thread, raster thread, thread for isolate, etc.

I wonder whether I can do it now? If not, I am willing to contribute!

**Comment by maheshmnj** [comment]
Hi @fzyzcjy, Thanks for filing the issue. I am not sure if this is possible as of today, Labeling this issue for further insights from the team.

cc: @chinmaygarde

**Comment by fzyzcjy** [comment]
Thanks. Looking forward to replies!

**Comment by ipr-fredrik2** [comment]
I would agree with this. Flutter's ecosystem having strict control over core affinity is a huge blocker for the development of any demanding application. We require strict control over hardware cores in order to properly optimize our application (which is already pushing boundaries on what's possible to do in realtime on mobile hardware). 

Flutter assigning all cores to what is essentially low-priority UI stuff (from our applications point of view) makes affinity optimization completely useless for us (we get worse responsivity and performance in our flutter App when we attempt to assign affinity for our hardworking threads).

Would also recommend to set the lowest frequency cores as efficiency cores, rather than the highest frequency cores as performance cores. But optimally I feel like the user (app developer) should be able to control whether or not affinity is enabled in flutter, or better yet, let the app developers set a "global" affinity mask that flutter is allowed to use, even if it just ends up being one or two hw cores.
